pkg/webhook/namespace/patching.go

```diff
@@ -0,0 +1,345 @@
+		for _, tnt := range tenants {
+			if strings.HasPrefix(ns.GetName(), fmt.Sprintf("%s-", tnt.GetName())) {
+				response := h.patchResponseForOwnerRef(tnt.DeepCopy(), ns, recorder)
+
+				return &response
+			}
+		}
+
+		response := admission.Denied("The Namespace prefix used doesn't match any available Tenant")
+
+		return &response
+	}
+
+	response := admission.Denied("Unable to assign namespace to tenant. Please use " + ln + " label when creating a namespace")
+
+	return &response
+}
+
+func (h *patchingHandler) patchResponseForOwnerRef(tenant *capsulev1beta2.Tenant, ns *corev1.Namespace, recorder record.EventRecorder) admission.Response {
+	scheme := runtime.NewScheme()
+	_ = capsulev1beta2.AddToScheme(scheme)
+	_ = corev1.AddToScheme(scheme)
+
+	o, err := json.Marshal(ns.DeepCopy())
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	if err = controllerutil.SetOwnerReference(tenant, ns, scheme); err != nil {
+		recorder.Eventf(tenant, corev1.EventTypeWarning, "Error", "Namespace %s cannot be assigned to the desired Tenant", ns.GetName())
+
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	recorder.Eventf(tenant, corev1.EventTypeNormal, "NamespaceCreationWebhook", "Namespace %s has been assigned to the desired Tenant", ns.GetName())
+
+	c, err := json.Marshal(ns)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	return admission.PatchResponseFromRaw(o, c)
+}
+
+func (h *patchingHandler) listTenantsForOwnerKind(ctx context.Context, ownerKind string, ownerName string, clt client.Client) (*capsulev1beta2.TenantList, error) {
+	tntList := &capsulev1beta2.TenantList{}
+	fields := client.MatchingFields{
+		".spec.owner.ownerkind": fmt.Sprintf("%s:%s", ownerKind, ownerName),
+	}
+	err := clt.List(ctx, tntList, fields)
+
+	return tntList, err
+}
+
+func (h *patchingHandler) validateNamespacePrefix(ns *corev1.Namespace, tenant *capsulev1beta2.Tenant) *admission.Response {
+	// Check if ForceTenantPrefix is true
+	if tenant.Spec.ForceTenantPrefix != nil && *tenant.Spec.ForceTenantPrefix {
+		if !strings.HasPrefix(ns.GetName(), fmt.Sprintf("%s-", tenant.GetName())) {
+			response := admission.Denied(fmt.Sprintf("The Namespace name must start with '%s-' when ForceTenantPrefix is enabled in the Tenant.", tenant.GetName()))
+
+			return &response
+		}
+	}
+
+	return nil
+}
+
+func (h *patchingHandler) syncNamespaceCordonLabel(ctx context.Context, c client.Client, ns corev1.Namespace) error {
+	tnt := &capsulev1beta2.Tenant{}
+
+	ln, err := capsuleutils.GetTypeLabel(tnt)
+	if err != nil {
+		return err
+	}
+
+	if label, ok := ns.Labels[ln]; ok {
+		if err = c.Get(ctx, types.NamespacedName{Name: label}, tnt); err != nil {
+			admission.Errored(http.StatusInternalServerError, err)
+		}
+	}
+
+	if !tnt.Spec.Cordoned {
+		delete(ns.Labels, "projectcapsule.dev/cordoned")
+	} else {
+		ns.Labels["projectcapsule.dev/cordoned"] = "true"
+	}
```

Comment:
only handle the true case at admission. deletion should be done by the controller.